### PR TITLE
Callback to future with failure handling

### PR DESCRIPTION
**Pull request by zgagnon**
Previously, callbacks were converted to futures by mapping a promise completion. This would not complete if the callback failed. Now promise is completed using both a map and a recover, allowing failed callbacks to be converted into failing futures.

### CODE REVIEW

File: colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala [hunk 1]
```diff
@@ -310,6 +310,23 @@ class CallbackSpec extends ColossusSpec {
...
+      value must equal(5)
+    }
+
+    "toFuture failed" in {
+      import scala.concurrent.ExecutionContext.Implicits.global
+
+      val badfunc = {f: (Try[Int] => Unit) => f(Failure[Int](new Exception))}
+      val callback = Callback(badfunc).map { i => i }
+      val result = Await.ready(callback.toFuture, 2 seconds).value.get
+      result mustBe a [Failure[Int]]
```
Comment (nsauro) [inline colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala]: This is throwing a compiler warning because of type erasure

Comment (DanSimon) [inline colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala]: yeah I've run into the same problem before, better to do `result.isFailure must equal(true)`

File: colossus/src/main/scala/colossus/service/Callback.scala [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package colossus
+jpackage colossus
```
Comment (nsauro) [inline colossus/src/main/scala/colossus/service/Callback.scala]: compiler error

File: colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala (hunk 1 above)
Comment (zgagnon) [inline colossus-tests/src/test/scala/colossus/service/CallbackSpec.scala]: coolbeans will do.

**Comment by DanSimon** [comment]
:+1:

**Comment by nsauro** [comment]
looks good

**Pull request merged**